starknet_py/cairo/serialization/function_serialization_adapter.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import List, Tuple
+
+from starknet_py.cairo.felt import CairoData
+from starknet_py.cairo.serialization.data_serializers.payload_serializer import (
+    PayloadSerializer,
+)
+from starknet_py.cairo.serialization.errors import InvalidTypeException
+from starknet_py.utils.tuple_dataclass import TupleDataclass
+
+
+@dataclass
+class FunctionSerializationAdapter:
+    """
+    Class serializing *args and **kwargs  by adapting them to function inputs.
+    """
+
+    inputs_serializer: PayloadSerializer
+    outputs_deserializer: PayloadSerializer
+
+    expected_args: Tuple[str] = field(init=False)
+
+    def __post_init__(self):
+        self.expected_args = tuple(self.inputs_serializer.serializers.keys())
+
+    def serialize(self, *args, **kwargs) -> CairoData:
+        """
+        Method using args and kwargs to match members and serialize them separately.
+
+        :return: Members serialized separately in SerializedPayload.
+        """
+        named_arguments = dict(kwargs)
+
+        if len(args) > len(self.expected_args):
+            raise InvalidTypeException(
+                f"Provided {len(args)} positional arguments, {len(self.expected_args)} possible."
+            )
```

Comment:
This could also be moved to separate method, but it's not necessary.

Reply:
Moved all checks to separate methods